tests/integration/test_log_rotation.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Feature: Logs are rotated via logrotate.d configuration."""
+
+import pathlib
+
+import jubilant
+from helpers import PATH_EXCLUDE
+
+from constants import INTERNAL_TELEMETRY_LOG_FILE
+
+LOG_DIR = str(pathlib.Path(INTERNAL_TELEMETRY_LOG_FILE).parent)
+
+
+async def test_deploy(juju: jubilant.Juju, charm: str):
+    # GIVEN an OpenTelemetry Collector charm and a principal
+    ## NOTE: /var/log/cloud-init.log and /var/log/cloud-init-output.log are always present
+    juju.deploy(charm, app="otelcol", config={"path_exclude": PATH_EXCLUDE})
```

Comment:
We don't need `PATH_EXCLUDE` anymore since we do not have the recursion problem. This would filter out irrelevant logs in the test, but adds clutter to the test.
```suggestion
    juju.deploy(charm, app="otelcol")
```